wire-runtime/src/test/java/com/google/protobuf/DescriptorProto.java

```diff
@@ -93,35 +66,35 @@ private DescriptorProto(Builder builder) {
   }
 
   @Override
-  public boolean equals(Object other) {
-    if (other == this) return true;
-    if (!(other instanceof DescriptorProto)) return false;
-    DescriptorProto o = (DescriptorProto) other;
-    return equals(name, o.name)
-        && equals(doc, o.doc)
-        && equals(field, o.field)
-        && equals(extension, o.extension)
-        && equals(nested_type, o.nested_type)
-        && equals(enum_type, o.enum_type)
-        && equals(extension_range, o.extension_range)
-        && equals(options, o.options);
+  protected void visitFields(Message.Visitor visitor) {
+    visitor.value(1, "name", name, TypeAdapter.STRING, false);
+    visitor.value(8, "doc", doc, TypeAdapter.STRING, false);
+    visitor.repeated(2, "field", field, FieldDescriptorProto.ADAPTER, false);
+    visitor.repeated(6, "extension", extension, FieldDescriptorProto.ADAPTER, false);
+    visitor.repeated(3, "nested_type", nested_type, DescriptorProto.ADAPTER, false);
+    visitor.repeated(4, "enum_type", enum_type, EnumDescriptorProto.ADAPTER, false);
+    visitor.repeated(5, "extension_range", extension_range, ExtensionRange.ADAPTER, false);
+    visitor.value(7, "options", options, MessageOptions.ADAPTER, false);
+    visitor.unknowns(this);
```

Comment:
Not sure what you're asking here. Who is the caller? And how would they handle unknowns?